apps/src/templates/instructions/codeReview/CodeReviewDataApi.js

```diff
@@ -39,11 +42,11 @@ export default class CodeReviewDataApi {
     });
   }
 
-  submitNewCodeReviewComment(commentText, token) {
+  submitNewCodeReviewComment(commentText) {
```

Comment:
should we return some sort of error if the token has not been set?

Reply:
Probably -- since we don't really handle not having a token at the moment, it felt out of scope for this change. Could be a follow-up?

Reply:
Can you create a task for that? I think we could do something similar to the [client api](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/clientApi.js#L485)

Reply:
Yeah, might be good to guard against not having the token since I think currently we do try to prevent service calls from happening in ReviewTab if token has not been set right? Alternatively, if there is a single API to get just the token, we could just grab the token on whichever service call is made first.

Reply:
It looks like we guard against rendering the review checkbox when we have no token, but mostly we do not have checks in place?

https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/instructions/ReviewTab.jsx#L268

Reply:
I'll look into it a bit :)

Reply:
Or perhaps rather than an error, have each function check if the token exists yet and if not, request it.

Reply:
My 2-cents on the in/out of scope: Since the ownership of the logic for getting the token is moving to CodeReviewDataApi, this feels like an appropriate time to handle error conditions surrounding the token. That way future developers of the ReviewTab don't need to be concerned about error conditions from the API that they are invoking. In an ideal world with infinite time, that would happen as part of this change.

Stepping out of the ideal world into the real world, I don't have an opinion around whether this needs to happen as part of this change or can be a follow-up. Deciding that would depend on how likely it is that one of these functions would be called before `getCodeReviewCommentsForProject`. If it's rather unlikely or pretty complicated, follow-up this sprint feels fine. If it's somewhat likely and not very complicated, it should probably happen as part of this change.

Reply:
^oops. I had an old version of this conversation cached. General thought process is the same, but ignore my conclusions :)

Reply:
It's probably more likely to be developer error vs user error if this ends up being called with no token (if we have some case in the future where we use this api elsewhere). It does make it more clear to the developer to see that they need to have cached the token before they call the function if we throw an error. On the user side if we throw an error, I'm guessing we should just see a generic save error?

Reply:
Got it, that makes sense. This pattern is heavily inspired by the topInstructionsDataApi, and it took me a while to figure out why my POST requests were failing when I added this originally :) I can save some future developer some time if I throw an error

Reply:
Added a check to throw if token isn't there for requests that require one (ie, non-GET). UI doesn't handle it gracefully (you just get Saving... spinner indefinitely if you try to create a comment without a token), but at least error message is descriptive?